pkg/validatorfactory/factory.go

```diff
@@ -196,15 +194,20 @@ func removeRefs(defs map[string]*spec.Schema, sch spec.Schema) spec.Schema {
 	// we squash it so that the Ref is direct without AllOf
 
 	if len(sch.AllOf) == 1 && len(sch.AllOf[0].Ref.String()) > 0 {
-		vCopy := sch
-		vCopy.AllOf = nil
-		vCopy.Default = nil
-		vCopy.Example = nil
-		vCopy.Description = ""
-
-		if reflect.DeepEqual(vCopy, spec.Schema{}) {
-			return removeRefs(defs, sch.AllOf[0])
+		vCopy := removeRefs(defs, sch.AllOf[0])
```

Comment:
For Test `error_multiple_fields` I get this error after this PR:

```diff
--- Expected
+++ Actual
@@ -13,4 +13,4 @@
	Status: (string) (len=7) "Failure",
-  Message: (string) (len=36) "failed to unmarshal document to YAML",
-  Reason: (v1.StatusReason) (len=10) "BadRequest",
+  Message: (string) (len=95) "Internal error occurred: yaml: unmarshal errors:\n  line 24: key \"containers\" already set in map",
+  Reason: (v1.StatusReason) (len=13) "InternalError",
	Details: (*v1.StatusDetails)({
@@ -23,3 +23,3 @@
		Type: (v1.CauseType) "",
-     Message: (string) (len=44) "line 24: key \"containers\" already set in map",
+     Message: (string) (len=70) "yaml: unmarshal errors:\n  line 24: key \"containers\" already set in map",
		Field: (string) ""
@@ -29,3 +29,3 @@
	}),
-  Code: (int32) 400
+  Code: (int32) 500
	}
```

As we can see the error is the same, but likely an unhandled type so it gets wrapped into an internal error. `errToStatus` likely needs to change to handle the error better:

https://github.com/kubernetes-sigs/kubectl-validate/blob/5aee348b797cc43baeac2791929133439b8eec36/pkg/cmd/validate.go#L91-L120

Reply:
#60 fixes this